xcat-inventory/xcclient/inventory/xcatobj.py

```diff
@@ -352,14 +368,37 @@ def validatevalue(self,objdict):
                         myval=''
                     myexpression=myexpression.replace('V{'+val+'}',"'"+str(myval).replace("'","\\'")+"'")                    
                 try:
-                    #print myexpression
                     evalexp=eval("lambda "+myexpression)
                     value=evalexp()
                 except Exception,e:                    
                     raise  InvalidValueException("Error: encountered some error when validate attribute ["+key+"] of object \""+self.name+"\": "+str(e)) 
                 if not value:
                     raise  InvalidValueException("Error: failed to validate attribute ["+key+"] of object \""+self.name+"\", criteria: \""+myexpression+"\"") 
 
+    @classmethod
+    def getfilerules(cls):
+        print yaml.dump(cls._files)
+
+    def getfilestosave(self):
+        filelist=[]
+        for key in self._files.keys():
+            depvallist=self._files[key]['file2savefilter']['depvallist']
+            expression=self._files[key]['file2savefilter']['expression'] 
+            for myexpression in expression:
+                for val in depvallist:
+                    myval=Util_getdictval(self._mydict,val)
+                    if myval is None:
+                        myval=''
+                    myexpression=myexpression.replace('V{'+val+'}',"'"+str(myval).replace("'","\\'")+"'")
+                try:
+                    evalexp=eval("lambda "+myexpression)
+                    value=evalexp()
+                except Exception,e:                    
+                    raise  InvalidValueException("Error: encountered some error when get the files to save in ["+key+"] of object \""+self.name+"\": "+str(e)) 
```

Comment:
Suggesting : `"Error: encountered some error when get the files to save in [%s] of object \"%s\": %s" % (key, self.name, str(e))`